sw/device/lib/irq.c

```diff
@@ -10,6 +10,8 @@ static const uint32_t IRQ_EXT_ENABLE_OFFSET = 11;
 static const uint32_t IRQ_TIMER_ENABLE_OFFSET = 7;
 static const uint32_t IRQ_SW_ENABLE_OFFSET = 3;
 
+void update_mtvec(char *ptr) { asm volatile("csrw mtvec, %0" ::"r"(ptr) :); }
```

Comment:
Isn't this more accurately a `uintptr_t`?  And unwrap from a single line.

Reply:
Yes, that's probably right. 

The formatting was done by clang-format, but it may change when I change the argument type.

Reply:
Apart of clarification, LGTM! It would be nice if we could get it in sooner rather than later, as some of the work that I am doing now requires this change.